src/test/scala/temple/builder/ServerBuilderTest.scala

```diff
@@ -0,0 +1,135 @@
+package temple.builder
+
+import org.scalatest.{FlatSpec, Matchers}
+import temple.ast.AttributeType._
+import temple.ast.Metadata.Database
+import temple.ast.{Attribute, AttributeType}
+import temple.detail.LanguageDetail.GoLanguageDetail
+import temple.generate.CRUD._
+import temple.generate.server.{CreatedByAttribute, IDAttribute, ServiceRoot}
+
+import scala.collection.immutable.ListMap
+
+class ServerBuilderTest extends FlatSpec with Matchers {
+
+  behavior of "ServerBuilderTest"
+
+  it should "build a correct simple ServiceRoot with all endpoints" in {
+    val serviceRoot: ServiceRoot = BuilderTestData.simpleTemplefile.servicesWithPorts.head match {
+      case (name, service, port) =>
+        ServerBuilder
+          .buildServiceRoot(
+            name,
+            service,
+            endpoints = Set(Create, Read, Update, Delete, List),
+            port = port.service,
+            detail = GoLanguageDetail("github.com/squat/and/dab"),
+          )
+    }
+    serviceRoot shouldBe ServiceRoot(
+      "test-service",
+      "github.com/squat/and/dab/test-service",
+      comms = Seq(),
+      port = 1024,
+      opQueries = ListMap(
+        Create -> "INSERT INTO test-service (id, bankBalance, name, isStudent, dateOfBirth, timeOfDay, expiry, image) VALUES ($1, $2, $3, $4, $5, $6, $7, $8) RETURNING id, bankBalance, name, isStudent, dateOfBirth, timeOfDay, expiry, image;",
```

Comment:
Now that the SQL-Gen has been updated to handle all of these, we should probs change these (and the other test SQL strings) to be generated through it, rather than using strings.

Reply:
I'd argue that calls out to the SQL code too much though and loses the atomicity of the unit tests and starts feeling more integration testy